### PR TITLE
LADSPA msvc compatibility

### PR DESCRIPTION
**Pull request by justnope**
* added FindFluidSynth.cmake
* update submodules
* Added _USE_MATH_DEFINES in cmake
* removed hardcode perl in cmake

I've realised that instead of editing ladspa.h, I could've also edited the cmake file so it exports all symbols like gcc & clang does. Since ladspa.h came from outside of lmms, there might be good reasons to not allow any changes to that file.

### CODE REVIEW

File: src/tracks/Pattern.cpp [hunk 1]
```diff
@@ -678,7 +678,7 @@ void PatternView::constructContextMenu( QMenu * _cm )
 
 	if( gui->pianoRoll()->currentPattern() &&
 			gui->pianoRoll()->currentPattern() != m_pat &&
-			not m_pat->empty() )
+			!m_pat->empty() )
```
Comment (PhysSong) [inline src/tracks/Pattern.cpp]: I think now it's time to revert this.

**Comment by PhysSong** [comment]
If the FluidSynth stuff is not necessary for this PR, I suggest making a separate request.

File: plugins/LadspaEffect/calf/CMakeLists.txt [hunk 2]
```diff
@@ -36,16 +36,25 @@ SET_TARGET_PROPERTIES(veal PROPERTIES PREFIX "")
...
 
 # Don't strip if "Debug" or "RelWithDebInfo"
 IF(LMMS_BUILD_WIN32 AND NOT CMAKE_BUILD_TYPE MATCHES "Deb")
 	ADD_CUSTOM_COMMAND(TARGET veal POST_BUILD COMMAND "${STRIP}" "$<TARGET_FILE:veal>")
 ENDIF()
 IF(NOT LMMS_BUILD_APPLE AND NOT LMMS_BUILD_OPENBSD)
-	SET_TARGET_PROPERTIES(veal PROPERTIES LINK_FLAGS "${LINK_FLAGS} -shared -Wl,-no-undefined")
+	IF(MSVC)
+		SET_TARGET_PROPERTIES(veal PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
+		TARGET_LINK_LIBRARIES(veal ${FLUIDSYNTH_LIBRARIES} Ws2_32)
```
[the remark this answers is not in the document]
Comment (justnope) [reply plugins/LadspaEffect/calf/CMakeLists.txt]: No idea why I included those libraries. I'll remove the entire block.
Do I need to link against m in non msvc builds like tap & swh or is it ok?

File: src/tracks/Pattern.cpp (hunk 1 above)
Comment (justnope) [reply src/tracks/Pattern.cpp]: I'll rebase on master, then it should be gone in this PR

File: plugins/LadspaEffect/calf/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (PhysSong) [reply plugins/LadspaEffect/calf/CMakeLists.txt]: There were no issues, so fine not to link against `libm`.

File: src/tracks/Pattern.cpp (hunk 1 above)
Comment (PhysSong) [reply src/tracks/Pattern.cpp]: Okay.

File: plugins/LadspaEffect/swh/CMakeLists.txt [hunk 3]
```diff
@@ -1,18 +1,26 @@
 # Create blank config.h
 FILE(WRITE ladspa/config.h "")
+INCLUDE(FindPerl)
+
+#TODO: should we do something when perl isn't found? IF(PERL_FOUND) ...
```
Comment (PhysSong) [inline plugins/LadspaEffect/swh/CMakeLists.txt]: I think we should print a warning message and `return()` in that case.

**Pull request closed without merging**